components/Chips/src/MDCChipField.m

```diff
@@ -0,0 +1,573 @@
+
+#pragma mark UIKeyInput
+
+- (void)deleteBackward {
+  [super deleteBackward];
+  if (self.text.length == 0) {
+    [self.deletionDelegate textFieldShouldRespondToDeleteBackward:self];
+  }
+}
+
+#if MDC_CHIPFIELD_PRIVATE_API_BUG_FIX && !(defined(__IPHONE_8_3) && (__IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_8_3))
+
+// WARNING: This is a private method, see the warning in MDCChipField.h.
+// This is only compiled if you explicitly defined MDC_CHIPFIELD_PRIVATE_API_BUG_FIX yourself, and
+// you are targeting an iOS version less than 8.3.
+- (BOOL)keyboardInputShouldDelete:(UITextField *)textField {
+  BOOL shouldDelete = YES;
+  if ([UITextField instancesRespondToSelector:_cmd]) {
+    BOOL (*keyboardInputShouldDelete)(id, SEL, UITextField *) =
+        (BOOL (*)(id, SEL, UITextField *))[UITextField instanceMethodForSelector:_cmd];
+    if (keyboardInputShouldDelete) {
+      shouldDelete = keyboardInputShouldDelete(self, _cmd, textField);
+      BOOL isIos8 = ([[[UIDevice currentDevice] systemVersion] intValue] == 8);
+      BOOL isLessThanIos8_3 = ([[[UIDevice currentDevice] systemVersion] floatValue] < 8.3f);
+      if (![textField.text length] && isIos8 && isLessThanIos8_3) {
+        [self deleteBackward];
+      }
+    }
+  }
+  return shouldDelete;
+}
+
+#endif
+
+#pragma mark - UIAccessibility
+
+- (CGRect)accessibilityFrame {
+  CGRect frame = [super accessibilityFrame];
+  return CGRectMake(frame.origin.x + self.textInsets.left,
+                    frame.origin.y,
+                    frame.size.width - self.textInsets.left,
+                    frame.size.height);
+}
+
+@end
+
+@interface MDCChipField ()
+    <MDCChipTextFieldDelegate, MDCTextInputPositioningDelegate, UITextFieldDelegate>
+@end
+
+@implementation MDCChipField {
+  NSMutableArray<MDCChipView *> *_chips;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    _chips = [NSMutableArray array];
+    _delimiter = MDCChipFieldDelimiterDefault;
+    _minTextFieldWidth = MDCChipFieldDefaultMinTextFieldWidth;
+    _contentEdgeInsets = MDCChipFieldDefaultContentEdgeInsets;
+    _showPlaceholderWithChips = YES;
+    _chipHeight = 32.0f;
+
+    MDCChipTextField *chipTextField = [[MDCChipTextField alloc] initWithFrame:self.bounds];
+    chipTextField.underline.hidden = YES;
+    chipTextField.delegate = self;
+    chipTextField.deletionDelegate = self;
+    chipTextField.positioningDelegate = self;
+    chipTextField.accessibilityTraits = UIAccessibilityTraitNone;
+    chipTextField.autocorrectionType = UITextAutocorrectionTypeNo;
+    chipTextField.autocapitalizationType = UITextAutocapitalizationTypeNone;
+    chipTextField.keyboardType = MDCChipFieldDefaultKeyboardType;
+    [chipTextField addTarget:self
+                      action:@selector(textFieldDidChange)
+            forControlEvents:UIControlEventEditingChanged];
+    [self addSubview:chipTextField];
+    _textField = chipTextField;
+  }
+  return self;
+}
+
```

Comment:
add `initWithCoder:` to support Interface Builder and archiving?

Reply:
Done.